samples/dockerbuilds/ARM/Dockerfile

```
@@ -0,0 +1,100 @@
+RUN tar -xvf ./${TOOLCHAIN_ARM}.tar.xz
+
+#########################################
+# Download and Configure OpenSSL
+
+RUN wget https://github.com/openssl/openssl/archive/refs/tags/OpenSSL_1_1_1f.tar.gz \
+&& tar -xvf ./OpenSSL_1_1_1f.tar.gz \
+&& cd openssl-OpenSSL_1_1_1f \
+&& ./Configure linux-armv4 --prefix=${TOOLCHAIN_SYSROOT}/bin/ --openssldir=${OPENSSL_ROOT_DIR} shared \
+&& make CC=${TOOLCHAIN_BIN}/${TOOLCHAIN_PLATFORM}-gcc AR=${TOOLCHAIN_BIN}/${TOOLCHAIN_PLATFORM}-ar \
+&& make install \
+&& cd ..
+
+#########################################
+# Build curl
+
+RUN wget http://curl.haxx.se/download/curl-7.60.0.tar.gz \
+&& tar -xvf curl-7.60.0.tar.gz \
+&& cd curl-7.60.0 \
+&& ./configure --with-sysroot=${TOOLCHAIN_SYSROOT} --prefix=${TOOLCHAIN_SYSROOT}/${TOOLCHAIN_PLATFORM}/libc --target=${TOOLCHAIN_PLATFORM} --with-ssl --with-zlib --host=${TOOLCHAIN_PLATFORM} CC=${TOOLCHAIN_BIN}/${TOOLCHAIN_PLATFORM}-gcc AR=${TOOLCHAIN_BIN}/${TOOLCHAIN_PLATFORM}-ar \
+&& make \
+&& make install \
+&& cd ..
+
+#########################################
+#Build uuid
+
+RUN wget https://mirrors.edge.kernel.org/pub/linux/utils/util-linux/v2.32/util-linux-2.32-rc2.tar.gz \
+&& tar -xvf util-linux-2.32-rc2.tar.gz \
+&& cd util-linux-2.32-rc2/ \
+&& ./configure --prefix=${TOOLCHAIN_PREFIX} --with-sysroot=${TOOLCHAIN_SYSROOT} --target=${TOOLCHAIN_NAME} --host=${TOOLCHAIN_NAME} --disable-all-programs  --disable-bash-completion --enable-libuuid \
+&& make \
+&& make install \
+&& cd ..
+
+#########################################
+# Build Azure C SDK
+
+RUN git clone --recursive https://github.com/azure/azure-iot-sdk-c.git
+WORKDIR azure-iot-sdk-c
+RUN mkdir cmake
+WORKDIR cmake
+
+# Create a toolchain file on the fly
+RUN echo "SET(CMAKE_SYSTEM_NAME Linux)     # this one is important" > toolchain.cmake
+RUN echo "SET(CMAKE_SYSTEM_VERSION 1)      # this one not so much" >> toolchain.cmake
+RUN echo "SET(CMAKE_SYSROOT ${TOOLCHAIN_SYSROOT}/${TOOLCHAIN_PLATFORM}/libc)" >> toolchain.cmake
+RUN echo "SET(CMAKE_C_COMPILER ${TOOLCHAIN_BIN}/${TOOLCHAIN_PLATFORM}-gcc)" >> toolchain.cmake
+RUN echo "SET(CMAKE_CXX_COMPILER ${TOOLCHAIN_BIN}/${TOOLCHAIN_PLATFORM}-g++)" >> toolchain.cmake
+RUN echo "SET(CMAKE_FIND_ROOT_PATH ${TOOLCHAIN_SYSROOT}/${TOOLCHAIN_PLATFORM})" >> toolchain.cmake
+RUN echo "SET(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)" >> toolchain.cmake
+RUN echo "SET(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)" >> toolchain.cmake
+RUN echo "SET(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)" >> toolchain.cmake
+RUN echo "SET(OPENSSL_ROOT_DIR ${OPENSSL_ROOT_DIR})" >> toolchain.cmake
+RUN echo "SET(OPENSSL_CRYPTO_LIBRARY ${OPENSSL_ROOT_DIR}/libcrypto.so.1.1)" >> toolchain.cmake
+RUN echo "SET(OPENSSL_SSL_LIBRARY ${OPENSSL_ROOT_DIR}/libssl.so.1.1)" >> toolchain.cmake
+RUN echo "SET(OPENSSL_INCLUDE_DIR ${OPENSSL_ROOT_DIR}/include)" >> toolchain.cmake
+RUN cmake -DCMAKE_TOOLCHAIN_FILE=toolchain.cmake -Duse_default_uuid=ON -Duse_condition=OFF ..
+RUN make
+
```

```suggestion
RUN make
RUN make install

```

---

do you really want to do make install? the toolchain.cmake sets the find root dir by using CMAKE_FIND_ROOT_PATH. If you are building outside of a container on a x86 machine, you can overwrite the libs with ARM libs

---

Fair fair. Updating the suggestion to the below:

```suggestion
RUN make
RUN cmake --install . --prefix ${TOOLCHAIN_SYSROOT}/${TOOLCHAIN_PLATFORM}
```

The current problem (before this change), is that there is nowhere where the application references where to get the SDK. There was no installation step, so in that case, the behavior should usually be to do an `add_subdirectory(azure-iot-sdk-c)` to bring in the CMake build tree so it has access to the libraries and includes. So I added the install so CMake could find those in the installed location. Updating here to not squash user's base install.